src/useImage.tsx

```diff
@@ -52,30 +52,30 @@ export default function useImage({
   }
 
```

Comment:
could DRY this hook out with: 

`const { promise, cache, error, src } = cache[sourceKey]; `

Reply:
sorry, not sure I follow the suggestion?

Reply:
Just that you continually access `cache[sourceKey]` - but if you use some local variables, you won't need to repeat that so many times. "DRY" means "Don't Repeat Yourself". 

So could look something like this: 

```
export default function useImage({
  srcList,
  imgPromise = imagePromiseFactory({decode: true}),
  useSuspense = true,
}: useImageProps): {src: string | undefined; isLoading: boolean; error: any} {
  const [, setIsSettled] = useState(false)
  const sourceList = removeBlankArrayElements(stringToArray(srcList))
  const sourceKey = sourceList.join('')

  if (!cache[sourceKey]) {
    // create promise to loop through sources and try to load one
    cache[sourceKey] = {
      promise: promiseFind(sourceList, imgPromise),
      cache: 'pending',
      error: null,
    }
  }

  const { promise, cache, error, src } = cache[sourceKey]; 

  // when promise resolves/reject, update cache & state
  if (cache === 'resolved') {
    return {src, isLoading: false, error: null}
  }

  if (cache === 'rejected') {
    if (useSuspense) throw error
    return {isLoading: false, error, src: undefined}
  }

  promise
    // if a source was found, update cache
    // when not using suspense, update state to force a rerender
    .then((src) => {
      cache[sourceKey] = {...cache[sourceKey], cache: 'resolved', src}
      if (!useSuspense) setIsSettled(sourceKey)
    })

    // if no source was found, or if another error occurred, update cache
    // when not using suspense, update state to force a rerender
    .catch((error) => {
      cache[sourceKey] = {...cache[sourceKey], cache: 'rejected', error}
      if (!useSuspense) setIsSettled(sourceKey)
    })

  // cache === 'pending')
  if (useSuspense) throw promise
  return {isLoading: true, src: undefined, error: null}
}
```

Reply:
Oh, I see what you're saying. DRY is mostly about reducing logic duplication - not property access. It could make the code slightly cleaner - or slightly harder to read. 

Will give it some thought. 

Regardless, seems out of scope for this PR. Thanks for the suggestions!